tools/src/main/java/org/apache/kafka/tools/ConnectPluginPath.java

```diff
@@ -0,0 +1,498 @@
+        }
+
+        @Override
+        public int hashCode() {
+            return Objects.hash(pluginLocation, className, type);
+        }
+    }
+
+    private static Set<Row> enumerateRows(PluginSource source, Map<String, Set<ManifestEntry>> manifests, PluginScanResult scanResult) {
+        Set<Row> rows = new HashSet<>();
+        // Perform a deep copy of the manifests because we're going to be mutating our copy.
+        Map<String, Set<ManifestEntry>> unloadablePlugins = manifests.entrySet().stream()
+                .collect(Collectors.toMap(Map.Entry::getKey, e -> new HashSet<>(e.getValue())));
+        scanResult.forEach(pluginDesc -> {
+            // Emit a loadable row for this scan result, since it was found during plugin discovery
+            rows.add(newRow(source, pluginDesc.className(), pluginDesc.type(), pluginDesc.version(), true, manifests));
+            // Remove the ManifestEntry if it has the same className and type as one of the loadable plugins.
+            unloadablePlugins.getOrDefault(pluginDesc.className(), Collections.emptySet()).removeIf(entry -> entry.type == pluginDesc.type());
+        });
+        unloadablePlugins.values().forEach(entries -> entries.forEach(entry -> {
+            // Emit a non-loadable row, since all the loadable rows showed up in the previous iteration.
+            // Two ManifestEntries may produce the same row if they have different URIs
+            rows.add(newRow(source, entry.className, entry.type, PluginDesc.UNDEFINED_VERSION, false, manifests));
+        }));
+        return rows;
+    }
+
+    private static Row newRow(PluginSource source, String className, PluginType type, String version, boolean loadable, Map<String, Set<ManifestEntry>> manifests) {
+        Set<String> rowAliases = new LinkedHashSet<>();
+        rowAliases.add(PluginUtils.simpleName(className));
+        rowAliases.add(PluginUtils.prunedName(className, type));
+        boolean hasManifest = manifests.containsKey(className);
+        return new Row(source.location(), className, type, version, new ArrayList<>(rowAliases), loadable, hasManifest);
+    }
+
+    private static void beginCommand(Config config) {
+        if (config.command == Command.LIST) {
+            listTablePrint(config,
+                    "pluginName",
+                    "firstAlias",
+                    "secondAlias",
+                    "pluginVersion",
+                    "pluginType",
+                    "isLoadable",
+                    "hasManifest",
+                    "pluginLocation" // last because it is least important and most repetitive
+            );
+        }
+    }
+
+    private static void handlePlugin(Config config, Row row) {
+        if (config.command == Command.LIST) {
+            String firstAlias = row.aliases.size() > 0 ? row.aliases.get(0) : "null";
+            String secondAlias = row.aliases.size() > 1 ? row.aliases.get(1) : "null";
```

Comment:
This isn't very human-readable, how about "N/A"?
```suggestion
            String firstAlias = row.aliases.size() > 0 ? row.aliases.get(0) : "N/A";
            String secondAlias = row.aliases.size() > 1 ? row.aliases.get(1) : "N/A";
```